src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetUnixVersion.cs

```diff
@@ -41,7 +42,7 @@ internal static string GetUnixVersion()
             {
                 fixed (byte* ptr = version)
                 {
-                    return new string((sbyte*)ptr);
+                    return Utf8StringMarshaller.ConvertToManaged(ptr)!;
```

Comment:
Calling `ConvertToManaged` on a pinned managed array feels odd.

Reply:
I think we should have some new internal helper that'd be similar to `strnlen` and have this be something like:
```cs
return Encoding.UTF8.GetString(version.AsSpan().TrimToNull());
```